spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -250,12 +282,40 @@ private List<ManifestFile> writeUnpartitionedManifests(
   private List<ManifestFile> writePartitionedManifests(
       ManifestContent content, Dataset<Row> manifestEntryDF, int numManifests) {
 
+    // Extract desired clustering/sorting criteria into a dedicated column
+    Dataset<Row> clusteredManifestEntryDF;
+    String clusteringColumnName = "__clustering_column__";
```

Comment:
nit: this can be a private static variable

Reply:
also if this is for sorting, should we call this `__sorting_column__`?

Reply:
I'm planning to reuse this same column for the more generic `Function` based approach. No strong opinion on naming convention though, or could use different names if you'd rather.